bom-alpha/build.gradle.kts

```diff
@@ -14,4 +14,6 @@ javaPlatform.allowDependencies()
 dependencies {
   // Add dependency on opentelemetry-bom to ensure synchronization between alpha and stable artifacts
   api(platform(project(":bom")))
-}
\ No newline at end of file
+
+  otelBom.addExtra("io.opentelemetry.semconv", "opentelemetry-semconv", "1.21.0-alpha")
```

Comment:
Should we include the new `io.opentelemetry.semconv:opentelemetry-semconv` artifact in the bom? I think yes.

Reply:
I think yes. Including it means a smaller impact (none?) to users, right?

Reply:
Not none because the artifact coordinates change from `io.opentelemetry:opentelemetry-semconv` to `io.opentelemetry.semconv:opentelemetry-semconv`.

There is some complexity with this resulting from the core component dependencies on semconv. For example, `opentelemetry-sdk-common` depends on semconv for resource attributes which are included by default. Suppose someone is using an old version of the sdk which relies on `io.opentelemetry.semconv:opentelemetry-semconv:1.21.0-alpha`, but separately wants to bring in a future semconv version `1.30.0-alpha` to produce new telemetry with new semantic conventions. And suppose between versions `1.21.0-alpha` and `1.30.0-alpha` we make breaking changes to the semconv code generation format which break `opentelemetry-sdk-common`. The user is essentially forced to either update to a later version of the SDK, or downgrade to a old version of semconv. 

My take away from this is:
- We should be careful about breaking changes to code generation logic in `io.opentelemetry.semconv:opentelemetry-semconv`, even while we still publish alpha artifacts and are allowed to make breaking changes
- We should try to produce a stable `io.opentelemetry.semconv:opentelemetry-semconv` sooner than later, to avoid the temptation / footgun of making breaking changes

Reply:
We spoke in the 9/7 Java SIG about this and concluded that we want to completely decouple this repository from `semantic-conventions-java`. This means inlining all current references to semantic conventions, and _not_ including `io.opentelemetry.semconv:opentelemetry-semconv` in the bom.

There's too much opportunity for version conflict given the instability of `io.opentelemetry.semconv:opentelemetry-semconv` and the likelihood of breaking changes, either stemming from changes to code gen logic or from changes to `semantic-conventions`.

Reply:
We are planning to include the semconv module in the instrumentation bom though.

Reply:
> We are planning to include the semconv module in the instrumentation bom though.

Playing devil's advocate here... Wouldn't that open the dor for inconsistencies and introduce further instabilities in the alpha bom?
I say that because I import both... always.

Reply:
The instrumentation bom is different because the instrumentation apis are actually coupled to the conventions of a particular version. Its not reasonable to use a version of the instrumentation API built for a particular version of semantic conventions with a different version.